### PR TITLE
decouple controller task bo from agent

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
- refactor task bo to decouple  from agent
## Modules
- [ ] UI
- [x] Controller
- [x] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [ ] add unit test
- [x] add necessary doc

### CODE REVIEW

**Review by dreamlandliu — DISMISSED** [review]
LGTM

File: server/controller-sdk/src/main/java/ai/starwhale/mlops/api/protocol/report/resp/TaskTrigger.java [hunk 1]
```diff
@@ -25,9 +25,9 @@
 public class TaskTrigger {
 
     /**
-     * task meta info
+     * unique id for the task
      */
-    private Task task;
+    Long id;
```
Comment (goldenxinxing) [inline server/controller-sdk/src/main/java/ai/starwhale/mlops/api/protocol/report/resp/TaskTrigger.java:30]: is jobinfo  necessary send to agent？so agent can upload task's result with common path of jobid

Comment (anda-ren) [reply server/controller-sdk/src/main/java/ai/starwhale/mlops/api/protocol/report/resp/TaskTrigger.java:30]: every task has it's result path: Task trigger.resultPath

**Review by dreamlandliu — APPROVED** [review]
LGTM

**Pull request merged**